src/GenSymIO.chpl

```diff
@@ -644,6 +644,168 @@ module GenSymIO {
+                   lineNumber=getLineNumber(),
+                   routineName=getRoutineName(), 
+                   moduleName=getModuleName(),
+                   errorClass="ParquetBuildError");
+        } else {
+            use Parquet;
+            var repMsg: string;
+            // May need a more robust delimiter then " | "
+            var (strictFlag, ndsetsStr, nfilesStr, allowErrorsFlag, arraysStr) = payload.splitMsgToTuple(5);
+            var strictTypes: bool = true;
+            if (strictFlag.toLower().strip() == "false") {
+              strictTypes = false;
+            }
+
+            var allowErrors: bool = "true" == allowErrorsFlag.toLower(); // default is false
+            if allowErrors {
+                gsLogger.warn(getModuleName(), getRoutineName(), getLineNumber(), "Allowing file read errors");
+            }
+
+            // Test arg casting so we can send error message instead of failing
+            if (!checkCast(ndsetsStr, int)) {
+                var errMsg = "Number of datasets:`%s` could not be cast to an integer".format(ndsetsStr);
+                gsLogger.error(getModuleName(), getRoutineName(), getLineNumber(), errMsg);
+                return new MsgTuple(errMsg, MsgType.ERROR);
+            }
+            if (!checkCast(nfilesStr, int)) {
+              var errMsg = "Number of files:`%s` could not be cast to an integer".format(nfilesStr);
+              gsLogger.error(getModuleName(), getRoutineName(), getLineNumber(), errMsg);
+              return new MsgTuple(errMsg, MsgType.ERROR);
+            }
+
+            var (jsondsets, jsonfiles) = arraysStr.splitMsgToTuple(" | ",2);
+            var ndsets = ndsetsStr:int; // Error checked above
+            var nfiles = nfilesStr:int; // Error checked above
+            var dsetlist: [0..#ndsets] string;
+            var filelist: [0..#nfiles] string;
+
+            try {
+                dsetlist = jsonToPdArray(jsondsets, ndsets);
+            } catch {
+                var errorMsg = "Could not decode json dataset names via tempfile (%i files: %s)".format(
+                                                   1, jsondsets);
+                gsLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR);
+            }
+
+            try {
+                filelist = jsonToPdArray(jsonfiles, nfiles);
+            } catch {
+                var errorMsg = "Could not decode json filenames via tempfile (%i files: %s)".format(nfiles, jsonfiles);
+                gsLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR);
+            }
+
+            var dsetdom = dsetlist.domain;
+            var filedom = filelist.domain;
+            var dsetnames: [dsetdom] string;
+            var filenames: [filedom] string;
+            dsetnames = dsetlist;
+
+            if filelist.size == 1 {
```

Comment:
Is it possible for filelist.size to ever equal zero?  We test for 1 here (and then check to see if it is blank) or else we assume it is greater than 1.

Reply:
I suspect not -- this code was directly copy/pasted from the HDF5 code (see line 442), so since the HDF5 code seems to work just fine, I would expect this identical code to behave in the same way